src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java

```diff
@@ -111,6 +114,20 @@ public DeploymentResult call() throws InterruptedException {
                     .kv("Deployment service config", deploymentServiceConfig.toPOJO().toString())
                     .log("Starting deployment task");
 
+            // Enforce deployment sequence validation using timestamps:
+            // If we have already processed a more recent deployment for a thing group, then we will drop
+            // this deployment and only process it if it is newer (or same) deployment for that thing group.
+            if (isStaleDeployment(deploymentDocument)) {
+                logger.atInfo().setEventType(DEPLOYMENT_TASK_EVENT_TYPE)
+                        .kv("Deployment service config", deploymentServiceConfig.toPOJO().toString())
+                        .log("Nucleus has a newer deployment for '{}' target. Rejecting the deployment.",
```

Comment:
logs should not end with a period, we already append a period to all logs.

Reply:
What do we know about the newer deployment? We should say what we do know, such as the timestamp or the configuration arn of the newer version that's currently applied.

Reply:
What should we log here for the customers? DeploymentID and configurationArn would that be ok? Do we need anything else?

Reply:
Config arn for the deployment we're rejecting and the config arn for the deployment that we're keeping, I guess?